src/sv-pipeline/04_variant_resolution/scripts/StitchFragmentedCNVs.java

```diff
@@ -0,0 +1,242 @@
+ * samples have identical genotypes, we can replace the first one by adding on the interval covered
+ * by the second one.
+ */
+public class StitchFragmentedCNVs {
+    private static final VCFParser.ByteSequence END = new VCFParser.ByteSequence("END");
+    private static final VCFParser.ByteSequence SVLEN = new VCFParser.ByteSequence("SVLEN");
+    private static final double PAD_FACTOR = .2;
+    private static final int MAX_PAD = 200000;
+    private static final String GZ = ".gz";
+
+    public static void main( final String[] args ) {
+        try ( final OutputStream os = createOutputStream(args[1]) ) {
+            try ( final VCFParser vcfParser = new VCFParser(args[0]) ) {
+                while ( vcfParser.hasMetadata() ) {
+                    vcfParser.nextMetaData().write(os);
+                }
+                final StitchableIterator sItr = new StitchableIterator(vcfParser);
+                VCFParser.Record stitchableRecord;
+                while ( (stitchableRecord = sItr.nextSubject(os)) != null ) {
+                    findExtension(stitchableRecord, sItr);
+                    stitchableRecord.write(os);
+                }
+            }
+        } catch ( final IOException ioe ) {
+            throw new VCFParser.MalformedVCFException("can't write output to " + args[1], ioe);
+        }
+    }
+
+    private static void findExtension( final VCFParser.Record stitchable,
+                                       final StitchableIterator sItr ) throws IOException {
+        final int startPos = stitchable.getPosition();
+        final VCFParser.ByteSequence endField = stitchable.getInfoField(END);
+        final int originalEndPos = endField.asInt(); // can't be null -- checked in isStitchable
+        int endPos = originalEndPos;
+        int pad = Math.min(MAX_PAD, (int)(PAD_FACTOR * (endPos - startPos)));
+        final int padStartPos = Math.max(1, startPos - pad);
+        int padEndPos = endPos + pad;
+
+        // sItr.hasNext returns false at EOF, or when the next record is too far away to overlap the subject
+        while ( sItr.hasNext() ) {
+            final VCFParser.Record record = sItr.next();
+            final int startPos2 = record.getPosition();
+            final int endPos2 = record.getInfoField(END).asInt(); // can't be null -- checked in isStitchable
+            final int pad2 = Math.min(MAX_PAD, (int)(PAD_FACTOR * (endPos2 - startPos2)));
+            final int padStartPos2 = Math.max(1, startPos2 - pad2);
+            final int padEndPos2 = endPos2 + pad2;
+            final int overlap = endPos - startPos2; // this seems clearly wrong. need clarification.
+            if ( overlap < pad && overlap < pad2 &&
+                    Math.max(padStartPos, padStartPos2) < Math.min(padEndPos, padEndPos2) &&
+                    genotypesMatch(stitchable, record) ) {
+                endPos = Math.max(endPos, endPos2);
+                pad = Math.min(MAX_PAD, (int)(PAD_FACTOR * (endPos - startPos)));
```

Comment:
I think an argument could be made that once you've glued two events together because you think they're actually the same event, that you can just treat it as a larger event.  But you're right, that that's not the way the current script works.
I'll see if I can figure out how to retain the original padding.